src/osx/carbon/font.cpp

```diff
@@ -1171,8 +1194,10 @@ wxString wxNativeFontInfo::ToString() const
             xml.Replace("\t",wxEmptyString,true);
             xml = xml.Mid(xml.Find("<plist"));
 
-            s.Printf("%d;%d;%d;%d;%s",
-                 2, // version
+            s.Printf("%d;%d;%d;%d;%d;%d;%s",
+                 3, // version
```

Comment:
The comment above needs to be updated to explain the v3 format rather than v2 which is not used any longer. In particular, I'd really like to know why do we include style but not weight into it now (and while I have some conjectures about the family, I'd appreciate having an explanation for it too).

Reply:
The font tests showed that a font created from a font info string was not getting family nor italic round-trip correct. I was exporting a string from the native font descriptor as xml and had to add what was missing in front (v2), now the tests showed that two other properties were not roundtrip correct. 

Eg a font could have a slant variant natively (thus expressed in the descriptor) or emulated by wx (normal font in font descriptor, slant matrix applied when constructing the font itself, 3rd param in `CTFontCreateWithFontDescriptor`). 

Deducing the font family does not work the way I thought it would, apparently the stylistic font class information is not always available. At least not from the descriptor alone. So while I try to find out whether I can make this work natively, exposing the family like this makes test pass correctly.

Reply:
I don't think making the tests pass under Mac is the highest priority -- it would be great if they passed, of course, but what we really need is for things to work correctly and I'm not totally sure if these changes actually achieve this. I.e. if it's just a temporary hack to make the tests pass, I'd rather avoid changing the format and add Mac-specific checks to the tests instead.

In response to the last question, I think `wxNativeFontInfo` should only be available when the corresponding native font is available, otherwise what's the point of using it? The rationale for it is to be able to exactly preserve, without any loss due to round tripping between it and wx font description, any native font. As long as we don't have any native font, we shouldn't need `wxNativeFontInfo` at all.

Because of this I'm also not sure at all if storing wx family in the serialized representation really makes sense...